### PR TITLE
Implementation of 'Distinct' command

### PR DESCRIPTION
**Pull request by viktortnk**
http://docs.mongodb.org/manual/reference/method/db.collection.distinct/

### CODE REVIEW

File: driver/src/main/scala/api/collections/genericcollection.scala [hunk 1]
```diff
@@ -175,6 +179,16 @@ trait GenericCollection[P <: SerializationPack with Singleton] extends Collectio
    */
   def count[H](selector: Option[pack.Document] = None, limit: Int = 0, skip: Int = 0, hint: Option[H] = None)(implicit h: H => CountCommand.Hint, ec: ExecutionContext): Future[Int] = runValueCommand(CountCommand.Count(query = selector, limit, skip, hint.map(h)))
 
+  /**
```
Comment (cchantep) [inline driver/src/main/scala/api/collections/genericcollection.scala]:
```scala
/**
 * Returns the distinct values for a specified field across a single collection and returns the results in an array.
 * @param key the field for which to return distinct values
 * @param selector the query selector that specifies the documents from which to retrieve the distinct values.
 */
def distinct[T](key: String, selector: Option[pack.Document] = None)(implicit ec: ExecutionContext, reader: pack.Reader[T]): Future[immutable.SortedSet[T]] = ???
```

**Comment by cchantep** [comment]
Hi, thanks for your contribution. It will be reviewed soon.

**Comment by cchantep** [comment]
Please squash the commit into a single one for the final review. Thanks

**Comment by viktortnk** [comment]
@cchantep Done

**Comment by cchantep** [comment]
Thanks @viktortnk

**Pull request merged**